test/coreneuron/unit/lfp/lfp.cpp

```diff
@@ -106,3 +108,77 @@ BOOST_AUTO_TEST_CASE(LFP_PointSource_LineSource) {
     nrnmpi_finalize();
 #endif
 }
+
+#ifdef ENABLE_SONATA_REPORTS
+BOOST_AUTO_TEST_CASE(LFP_ReportEvent) {
+    const std::string report_name = "compartment_report";
+    const std::vector<uint64_t> gids = {42, 134};
+    const std::vector<int> segment_ids = {0, 1, 2, 3, 4};
+    std::vector<double> curr(segment_ids.size());
+
+    NrnThread nt;
+    nt.mapping = new NrnThreadMappingInfo;
+    auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
+    // Generate mapinfo CellMapping
+    for (const auto& gid: gids) {
+        mapinfo->mappingvec.push_back(new CellMapping(gid));
```

Comment:
these are deleted?

Reply:
NrnThreadMappingInfo destructor takes care of it
```
struct NrnThreadMappingInfo {
    /** list of cells mapping */
    std::vector<CellMapping*> mappingvec;
    ...
    /** @brief memory cleanup */
    ~NrnThreadMappingInfo() {
        for (size_t i = 0; i < mappingvec.size(); i++) {
            delete mappingvec[i];
        }
    }
```
And I destroy mapinfo at the end
```
    nt.mapping = new NrnThreadMappingInfo;
    auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
    ...
    delete mapinfo;
```

Reply:
👍 forgot to check destructor !